joml-geometry/src/main/java/org/terasology/joml/geom/AABBd.java

```diff
@@ -281,6 +301,62 @@ public Vector3f center(Vector3f dest) {
         return dest.set(minX + ((maxX - minX) / 2.0f), minY + ((maxY - minY) / 2.0f), minZ + ((maxZ - minZ) / 2.0f));
     }
 
+    @Override
+    public AABBd expand(double dx, double dy, double dz, AABBd dest) {
+        return dest.set(
+            minX - dx,
+            minY - dy,
+            minZ - dz,
+            maxX + dx,
+            maxY + dy,
+            maxZ + dz
+        );
```

Comment:
This can lead to invalid AABB bounds, as it is neither checked here or in `.set`.

There should be a test for the correct behavior of this case.

Reply:
what would be the correct behavior in this case?

Reply:
I think this depends whether we want to keep `AABB`s closer to JOML behavior of `Vector`s or the behavior of our `BlockRegion`.

I tend to vote for keeping it closer to JOML by allowing _invalid_ AABBs, and having a check for that which can be used by a consumer of the type. The docs need to be adjusted to state that this can leave the AABB in an invalid state.